entgql/extension.go

```diff
@@ -231,6 +285,299 @@ func (e *Extension) mapScalar(f *gen.Field, op gen.Op) string {
 	return scalar
 }
 
+// genSchema generates the graphql schema
+func (e *Extension) genSchema() gen.Hook {
+	return func(next gen.Generator) gen.Generator {
+		var (
+			queries    = make(map[string]*ast.FieldDefinition)
+			interfaces = make(map[string]*ast.InterfaceDefinition)
+			scalars    = make(map[string]*ast.ScalarDefinition)
+			objects    = make(map[string]*ast.ObjectDefinition)
+			enums      = make(map[string]*ast.EnumDefinition)
+			inputs     = make(map[string]*ast.InputObjectDefinition)
+		)
+		return gen.GenerateFunc(func(g *gen.Graph) error {
+			nodes, err := filterNodes(g.Nodes)
+			if err != nil {
+				return err
+			}
+			if err := next.Generate(g); err != nil {
+				return err
+			}
+
+			addTimeScalarDefinition(scalars)
+			addNodeDefinition(interfaces)
```

Comment:
this is part of the relay specification

Reply:
Is this a question or a comment? The node definitions are always needed, whether using relay spec or not. The relay spec would simply wrap it into a `Node` rather than returning the type directly.

Reply:
My Mistake. I confused this with the `node(id: ID!)` query